mobilenet-burn/src/model/mobilenet.rs

```diff
@@ -0,0 +1,169 @@
+use alloc::vec;
+use alloc::vec::Vec;
+use burn::{
+    config::Config,
+    module::Module,
+    nn::{
+        pool::{AdaptiveAvgPool2d, AdaptiveAvgPool2dConfig},
+        Dropout, DropoutConfig, Linear, LinearConfig,
+    },
+    tensor::{backend::Backend, Tensor},
+};
+
+use crate::model::{
+    conv_norm::Conv2dNormActivationConfig, inverted_residual::InvertedResidualConfig,
+    utils::make_divisble,
+};
+
+use super::{
+    conv_norm::{Conv2dNormActivation, NormalizationType},
+    inverted_residual::InvertedResidual,
+};
+
+#[derive(Debug, Module)]
+pub struct MobileNetV2<B: Backend> {
+    features: Vec<FeatureLayersType<B>>,
+    classifier: Vec<ClassifierLayersType<B>>,
+    avg_pool: AdaptiveAvgPool2d,
+}
+
+#[derive(Debug, Config)]
+pub struct MobileNetV2Config {
+    #[config(default = "1000")]
+    num_classes: usize,
+
+    #[config(default = "1.0")]
+    width_mult: f32,
+
+    #[config(default = "vec![vec![]]")]
+    inverted_residual_setting: Vec<Vec<usize>>,
+
+    #[config(default = "8")]
+    round_nearest: usize,
+
+    norm_layer: NormalizationType,
+
+    #[config(default = "0.2")]
+    dropout: f64,
+}
+#[derive(Module, Debug)]
+enum FeatureLayersType<B: Backend> {
+    Block(InvertedResidual<B>),
+    Conv(Conv2dNormActivation<B>),
+}
+#[derive(Module, Debug)]
+enum ClassifierLayersType<B: Backend> {
+    Dropout(Dropout),
+    Linear(Linear<B>),
+}
+impl MobileNetV2Config {
+    pub fn init<B: Backend>(&self, device: &B::Device) -> MobileNetV2<B> {
+        let input_channel = 32;
+        let last_channel = 1280;
+        let mut inverted_residual_setting = self.inverted_residual_setting.clone();
+        if inverted_residual_setting.is_empty() {
+            inverted_residual_setting = vec![
+                // t, c, n, s
+                vec![1, 16, 1, 1],
+                vec![6, 24, 2, 2],
+                vec![6, 32, 3, 2],
+                vec![6, 64, 4, 2],
+                vec![6, 96, 3, 1],
+                vec![6, 160, 3, 2],
+                vec![6, 320, 1, 1],
+            ]
+        }
+        if inverted_residual_setting[0].len() != 4 {
+            panic!(
+                "inverted_residual_setting should be non-empty or a 4-element list, got {:#?}",
+                self.inverted_residual_setting
+            )
+        }
```

Comment:
This can probably just be a const array definition.

Reply:
Yes that can be done. It would also eliminate the need for the `panic!` in the next few lines